src/middlewares/insights.ts

```diff
@@ -83,10 +83,16 @@ aa('setUserToken', 'your-user-token');
           // Context: The umd build of search-insights is asynchronously loaded by the snippet.
           //
           // When user calls `aa('setUserToken', 'my-user-token')` before `search-insights` is loaded,
-          // it is stored in `aa.queue` and we are reading it to set userToken to search call.
-          // This queue is meant to be consumed whenever `search-insights` is loaded and when it runs `processQueue()`.
-          // But the reason why we handle it here is to prevent the first search API from being triggered
-          // without userToken because search-insights is not loaded yet.
+          // ['setUserToken', 'my-user-token'] gets stored in `aa.queue`.
+          // Whenever `search-insights` is finally loaded, it will process the queue.
+          //
+          // But the reason why we handle it here is
+          // (1) At this point, even though `search-insights` is not loaded yet,
+          // we still want to read the token from the queue.
+          // Otherwise, the first search call will be fired without the token.
+          // (2) Or, user could use a customized client of `search-insights`,
+          // for example, by using `createInsightsClient` function.
+          // Then `processQueue` might not be called. But we still want to read the token from the queue.
```

Comment:
in that case, there's no queue, right? I think (2) should be removed

Reply:
Ah you're correct!